.github/workflows/deploy_gcp_redis.yaml

```diff
@@ -0,0 +1,68 @@
+name: Deploy Redis to GCP
+
+on:
+  push:
+    branches:
+      - main
+      - testing
+      - production
+    paths:
+      - ".github/workflows/deploy_gcp_redis.yaml"
+  workflow_dispatch:
+
+jobs:
+  DeployRedisToGCP:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: "read"
+      id-token: "write"
+
+    # TODO: replace improve-gcp-deploy with main
+    environment: gcp-${{ (github.ref_name == 'main' && 'testing') || github.ref_name }}
+
+    env:
+      RESOURCE_PREFIX: ${{ secrets.PROJECT_NAME }}-${{ (github.ref_name == 'main' && 'testing') || github.ref_name }}
```

Comment:
Just a note that this code + trigger conditions will change in #327  -- if this branch gets merged first, can you ping me to update #327 ? If #327 gets merged first, would you mind updating this script (copy-paste!) before merging?

(cc: @markbotterill  also pls let me know when you review/merge #327 !)